Add documentation for negated ignored files and add a test for it as well (#6155)
rust-lang/rustfmt: Configurations.md

ignore = ["/"]
```

> If you want to allow specific paths, use `!`:
> 
> ```toml
> ignore = ["bar_dir/*", "!bar_dir/*/what.rs"]
> ```
> 
> In this case, all files under `bar_dir` will be ignored, except files like `bar_dir/sub/what.rs`
> or `bar_dir/another/what.rs`.
> 

Reviewer: To be totally honest, I didn't even know this was possible. Nice to see that it's doable 😁! I might rephrase it slightly:

````suggestion
If you want to allow specific paths that would otherwise be ignored, prefix those paths with a `!`:

```toml
ignore = ["bar_dir/*", "!bar_dir/*/what.rs"]
```

In this case, all files under `bar_dir` will be ignored, except files like `bar_dir/sub/what.rs`
or `bar_dir/another/what.rs`.

````

Author: :+1:
